src/group_theory/group_action/fixing_subgroup.lean

```diff
@@ -0,0 +1,201 @@
+lemma forall₂_swap' {ι₁ ι₂ ι₃ : Sort*} {κ₁ : ι₁ → Sort*} {κ₂ : ι₁ → Sort*}
+  {p : Π i₁, κ₁ i₁ → Π i₂, κ₂ i₂ → ι₃ → Prop} :
+  (∀ i₁ j₁ i₂ j₂ i₃, p i₁ j₁ i₂ j₂ i₃) ↔ ∀ i₂ j₂ i₁ j₁ i₃, p i₁ j₁ i₂ j₂ i₃ :=
+⟨λ h i₂ j₂ i₁ j₁ i₃, h i₁ j₁ i₂ j₂ i₃, λ h i₁ j₁ i₂ j₂ i₃, h i₂ j₂ i₁ j₁ i₃⟩
+
+@[simp] lemma antitone_to_dual_comp_iff : antitone (to_dual ∘ f) ↔ monotone f :=
+iff.rfl
+
+@[simp] lemma monotone_comp_of_dual_iff : monotone (f ∘ of_dual) ↔ antitone f :=
+forall_swap
+
+@[simp] lemma antitone_on_to_dual_comp_iff : antitone_on (to_dual ∘ f) s ↔ monotone_on f s :=
+iff.rfl
+
+@[simp] lemma monotone_on_comp_of_dual_iff : monotone_on (f ∘ of_dual) s ↔ antitone_on f s :=
+-- forall₂_swap /- Does not work -/
+begin
+split; exact λ hf a ha b hb h, hf hb ha h
+end
+
+end antitonicity
+
+section monoid
+
+open mul_action
+
+variables (M : Type*) {α : Type*} [monoid M] [mul_action M α]
+
+/-- The submonoid fixing a set under a `mul_action`. -/
+@[to_additive /-" The additive submonoid fixing a set under an `add_action`. "-/]
+def fixing_submonoid (s : set α) : submonoid M :=
+{ carrier := { ϕ : M | ∀ x : s, ϕ • (x : α) = x },
+  one_mem' := λ _, one_smul _ _,
+  mul_mem' := λ x y hx hy z, by rw [mul_smul, hy z, hx z], }
+
+lemma mem_fixing_submonoid_iff {s : set α} {m : M} :
+  m ∈ fixing_submonoid M s ↔ ∀ (y : α) (hy : y ∈ s), m • y = y :=
+⟨λ hg y hy, hg ⟨y, hy⟩, λ h ⟨y, hy⟩, h y hy⟩
+
+variable (α)
+
+/-- The Galois connection between fixing submonoids and fixed points of a monoid action -/
+theorem fixing_submonoid_fixed_points_connection : galois_connection
+  (order_dual.to_dual ∘ (λ s : set α, fixing_submonoid M s))
+  ((λ P : submonoid M, (fixed_points P α)) ∘ order_dual.of_dual) :=
+λ s P, ⟨λ h s hs p, h p.2 ⟨s, hs⟩, λ h p hp s, h s.2 ⟨p, hp⟩⟩
+
+lemma fixing_submonoid_is_antitone : antitone (λ (s : set α), fixing_submonoid M s) :=
+galois_connection.monotone_l (fixing_submonoid_fixed_points_connection M α)
+
+lemma fixed_points_is_antitone :
+  antitone (λ (P : submonoid M), fixed_points P α) :=
+(fixing_submonoid_fixed_points_connection M α).monotone_u.dual_left
+
+/-- Fixing submonoid of union is intersection -/
+lemma fixing_submonoid_of_union {s t : set α} :
+  fixing_submonoid M (s ∪ t) = (fixing_submonoid M s) ⊓ (fixing_submonoid M t) :=
+galois_connection.l_sup (fixing_submonoid_fixed_points_connection M α)
+
+/-- Fixing submonoid of Union is intersection -/
+lemma fixing_submonoid_Union {ι : Sort*} {s : ι → set α} :
+  fixing_submonoid M (⋃ i, s i) = ⨅ i, fixing_submonoid M (s i) :=
+galois_connection.l_supr (fixing_submonoid_fixed_points_connection M α)
+
+/-- Fixed points of sup of submonoids is intersection -/
+lemma fixed_points_sup {P Q : submonoid M} :
+  fixed_points (P ⊔ Q) α = fixed_points P α ∩ fixed_points Q α :=
+  galois_connection.u_inf (fixing_submonoid_fixed_points_connection M α)
+
+/-- Fixed points of supr of submonoids is intersection -/
+lemma fixed_points_supr {ι : Sort*} {P : ι → submonoid M} :
```

Comment:
```suggestion
lemma fixed_points_submonoid_supr {ι : Sort*} {P : ι → submonoid M} :
```
So that you have a matching name available for the `subgroup` version.

Reply:
fixed_points_of_submonoid_supr

Reply:
There should be no `of` in the name, it's not a hypothesis.